subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy

```diff
@@ -68,6 +68,44 @@ class FasterIncrementalAndroidBuildsPerformanceTest extends AbstractCrossBuildPe
         results
     }
 
+    def "file system watching baseline non-abi change (build comparison)"() {
+        given:
+        runner.buildSpec {
+            displayName("with file system watching")
+            configureForNonParallel(delegate)
+            testProject.configureForNonAbiChange(delegate)
+            invocation {
+                args.addAll(Optimization.WATCH_FS.arguments)
+            }
+        }
+        runner.buildSpec {
+            displayName("with file system watching - Gradle 6.7")
+            configureForNonParallel(delegate)
+            testProject.configureForNonAbiChange(delegate)
+            invocation {
+                args.addAll(Optimization.WATCH_FS.arguments)
+                distribution(buildContext.distribution("6.7-rc-2"))
+            }
+        }
+        runner.buildSpec {
+            displayName("without file system watching")
+            configureForNonParallel(delegate)
+            testProject.configureForNonAbiChange(delegate)
+        }
+
+        when:
+        def results = runner.run()
+        then:
+        results
+    }
+
+    private static void configureForNonParallel(GradleBuildExperimentSpec.GradleBuilder builder) {
+        builder.invocation {
+            args.add("-Dorg.gradle.parallel=false")
+            args.add("-Dorg.gradle.workers.max=1")
+        }
+    }
```

Comment:
Let's add some documentation about why we are disabling parallelism here. Maybe some general explanation about what this test does and why is in order, too.

Reply:
I added some explanation in the method.